ecdsa/src/ecdsa.rs

```diff
@@ -277,12 +292,28 @@ mod tests {
                         s: s_assigned,
                     };
 
-                    let pk_in_circuit = ecc_chip.assign_point(ctx, self.public_key)?;
+                    let point = self.public_key.map(|point| ecc_chip.to_rns_point(point));
+                    let (x, y) = point
+                        .map(|point| (point.x().clone(), point.y().clone()))
+                        .unzip();
+                    let (x, y) = if self.valid_input {
+                        (x.clone(), y.clone())
+                    } else {
+                        // Generate a point that is not on the curve.
+                        (x.clone(), x.clone())
+                    };
+
+                    let (pk_in_circuit, is_pk_on_curve) = ecc_chip.assign_x_y(ctx, x.into(), y.into())?;
+                    let enable_skipping_invalid_signature = scalar_chip.assign_constant(ctx, (self.enable_skipping_invalid_signature as u64).into())?;
+                    let enable_skipping_invalid_signature = scalar_chip.is_not_zero(ctx, &enable_skipping_invalid_signature)?;
+                    scalar_chip.one_or_one(ctx, &enable_skipping_invalid_signature, &is_pk_on_curve)?;
```

Comment:
Do we need to check this because the public key is something the prover can generate on its own (I think?) so can always make sure the points are on the curve? Or is it somehow necessary for invalid signatures?